### DISCUSSION TITLE
Changing the Device Name in Enrollment

### DISCUSSION THREAD

**Discussion by gcortes** [ESPresense/ESPresense#826] [Q&A] [answered]
I tried enrolling my iPhone following the instruction on the Apple Devices page. In step 2, it say to fill in the name, but doesn't specify the name of what. I assumed it was the name for the ESP32 so I called it esp-kitchen. The enrollment worked and now my phone is called esp-kitchen in MQTT Explorer. I tried forgetting the device on my iPhone and enrolling under a different name. It didn't change the name. It's not the end of the world that it's called esp-kitchen since the name is used internally. Still, I'd like to change it. Is this possible?

I also noticed after linking a second ESP32 that both devices are listed as ESPrense. Is there a way to give the a unique name?

To make matters worse, I rebooted the esp32 devices to effect some unrelated changes. Now the iPhone doesn't appear in MQTT Explorer for the devices I rebooted. I then re enrolled the phone and it reappeared in MQTT Explorer. Do I have to re-enroll every time I reboot the esp32?

After letting the phone sit for an hour or two, esp-kitchen stopped reporting. I re-enrolled it and it started to work again. I then rebooted the phone in the hopes of clearing things up. The esp-kitchen stopped reporting again. 

I did notice another MQTT Explorer entry labeled "irk:34cc...". Under it are entries for the three esp32's that I enrolled. I'm going to try using this in Home Assistant.

**Comment by tenly2000** [comment, marked as answer]
I'm only 2 days ahead of you, so I may be wrong, but:

#1:  You can rename the ESP32 at any time from it's web interface.  The setting is called "Room".

<img width="562" alt="Screen Shot 2023-03-06 at 9 36 56 PM" src="https://user-images.githubusercontent.com/33942031/223305850-87fcc850-f7c9-4d83-a585-b5d8974d7b1b.png">

#2:  I think you only need to pair your iPhone's to 1 sensor, 1 time (and you don't even need to leave it paired)
  - Pairing it seems to create an association between your device's IRK and the name you paired it with.  You can see this with MQTT Explorer under the \ESPresence\Settings topic.  Each device you pair will have an IRK associated with an "id".  It's the "id" that you need to configure in HomeAssistant for MQTT_ROOM
  - Optionally, you may want to use the FILTER configuration item on EACH ESP32 configuration page to ONLY send data about the handful of ID's that you care about (separate the ID's with spaces).
- Make sure you "Save" (bottom right) and "Restart" (top left) after making changes to the ESP32 configuration. (I don't know that it's required, I just know it might be - and it doesn't hurt)
 
<img width="565" alt="Screen Shot 2023-03-06 at 9 37 32 PM" src="https://user-images.githubusercontent.com/33942031/223305878-7a151602-0012-4899-80c3-13526cfdacd3.png">

#3:  The ESP32's should be auto-discovered by the MQTT integration in Home Assistant as soon as you set the MQTT parameters on the ESP32.  You have to add each device that you want to track to the configuration.yaml and specify either the "id" of the device or its "irk:" number.

**Reply by gcortes** [reply]
It looks like I have everything working now.

https://github.com/ESPresense/ESPresense/issues/1: The name I was referring to was the one that appeared on the Bluetooth page in iOS. All my esp32 devices already had unique names as shown in your example. The issue is essentially moot.

https://github.com/ESPresense/ESPresense/issues/2: I agree with your observation about only needing to pair/enroll one device. I did three. Two of the ones that I didn't do correctly showed the presence of the iPhone under the IRT in MQTT Explored when in proximity.

https://github.com/ESPresense/ESPresense/pull/3: My esp32 devices were already discovered by Home Assistant before I started setting up the IRT. I already had a sensor configured in the configuration.yaml using the generic id for my phone. I changed it to use the IRT and it's been working since yesterday.

My problem was because I used the name entered in the enrollment as the id for the phone. The name initially appeared in MQTT Explorer and then disappeared as described in my initial post. There are still two entries on the Bluetooth panel in iOS settings. Both are disconnected. They disconnect immediately after first connecting when paired. I agree with you that they may not even have to be there, certainly not more than one.

Were I to do it over again, the solution in my case would be to enroll one esp32, look in MQTT explorer for a new IRT, and then use that IRT in the configuration.yaml to define the sensor. The IRT can be verified by moving close and away from an esp32 to see if the IRT in question changes. The enrollment name is far too unreliable in my installation.

**Comment by tenly2000** [comment]
@gcortes 

> After letting the phone sit for an hour or two, esp-kitchen stopped reporting. I re-enrolled it and it started to work again. I then rebooted the phone in the hopes of clearing things up. The esp-kitchen stopped reporting again.

This was also happening to me on one of my 6 ESP32's.  It turned out that it was the USB power adapter I had plugged it into.  I guess it was old and underpowered.  I replaced it with one from an old iPhone and it's been rock solid ever since.
